app/controllers/instructor/sections_controller.rb

```diff
@@ -22,10 +22,11 @@ def update
 	def require_authorizec_for_current_section
 		if current_section.course.user != current_user
 			render text: 'Unauthorized', status: :unauthorized
+		end
 	end
 
 	def current_section
-		@current_section ||= Lesson.find(params[:id])
+		@current_section ||= Section.find(params[:id])
```

Comment:
When loading the `current_section` we need to use the `Section` model.  Not the `Lesson` one.